### ISSUE TITLE
Wrong DETECTED_CATTLE_AGENT_IP in HA deployment

### ISSUE THREAD

**Issue by MihaiAnei** [rancher/rancher#5867]
**Rancher Version:**
`v1.1.3`
**Docker Version:**
```
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:        Thu Aug 18 05:33:38 2016
 OS/Arch:      linux/amd64
```
**OS and where are the hosts located? (cloud, bare metal, etc):**
```
--OS--
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"
NAME="Ubuntu"
VERSION="16.04.1 LTS (Xenial Xerus)"
KERNEL="4.4.0-36-generic"
--Location--
VMs in a vCenter Cluster on local internal network
```
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
```
HA rancher with HAProxy manually configured to balance the traffic 
on 80 and 443 between the 3 hosts in HA. 
External mysql database on a different server.
The server uses AD for authentication
```
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
`Cattle`
**Steps to Reproduce:**
After the HA deployment is successful, create a new environment and add new hosts. The Host Registration URL is the domain name of the haproxy.
**Results:** 
The hosts are added in the environment with the correct hostname but ALL with the ip of the haproxy. Debugging the agents on the host I can see that DETECTED_CATTLE_AGENT_IP is set with the IP of the haproxy.
**Expected:**
DETECTED_CATTLE_AGENT_IP is set with the correct IP of the hosts and in Rancher UI you can see the correct IP.
**Workaround:**
When adding the custom hosts set CATTLE_AGENT_IP to the actual IP of the hosts.

I realised that there is a problem when trying to add Convoy NFS and the containers remained in Initialising state. You may think of other implications if the IP of the hosts are wrong in rancher.

**Comment by vincent99** [comment]
This sounds like you're missing sending X-Forwarded-For and/or PROXY Protocol... The source IP the server container sets for the request IS the HAProxy IP

**Comment by manei** [comment]
Oh, I see. I thought the mechanism of detecting the host`s IP is not depended on that. If so, it seems like I need to change the haproxy configuration. 
On that topic, do you guys think you can put in the documentation a recommended haproxy configuration for 80 and 443 (SSL passthrough) ports? For example, right now I'm using layer 4 load balancing with mode tcp, which is working pretty well but I have this problem.
Thank you @vincent99

**Comment by MihaiAnei** [comment]
I'll add here a configuration that works for me, maybe it helps someone.

```
global
	log /dev/log	local0
	log /dev/log	local1 notice
	chroot /var/lib/haproxy
	stats socket /run/haproxy/admin.sock mode 660 level admin
	stats timeout 30s
	user haproxy
	group haproxy
	daemon

	# Default SSL material locations
	ca-base /etc/ssl/certs
	crt-base /etc/ssl/private

	# Default ciphers to use on SSL-enabled listening sockets.
	# For more information, see ciphers(1SSL). This list is from:
	#  https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/
	ssl-default-bind-ciphers ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS
	ssl-default-bind-options no-sslv3

defaults
	defaults
  	mode http
  	log global
  	option httplog
  	option  http-server-close
  	option  dontlognull
  	option  redispatch
  	option  contstats
  	retries 3
  	backlog 10000
  	timeout client          25s
	timeout connect          5s
  	timeout server          25s
	# timeout tunnel available in ALOHA 5.5 or HAProxy 1.5-dev10 and higher
  	timeout tunnel        3600s
 	timeout http-keep-alive  1s
  	timeout http-request    15s
  	timeout queue           30s
  	timeout tarpit          60s
  	default-server inter 3s rise 2 fall 3
 	option forwardfor
	errorfile 400 /etc/haproxy/errors/400.http
	errorfile 403 /etc/haproxy/errors/403.http
	errorfile 408 /etc/haproxy/errors/408.http
	errorfile 500 /etc/haproxy/errors/500.http
	errorfile 502 /etc/haproxy/errors/502.http
	errorfile 503 /etc/haproxy/errors/503.http
	errorfile 504 /etc/haproxy/errors/504.http

frontend rancher80
    bind *:80 name http
    stats uri /haproxy?stats
    maxconn 10000
    default_backend nodes80

frontend rancher443
    bind *:443 ssl crt /etc/ssl/private/<certifiacte-you-used-at-script-creation-time>.pem
    stats uri /haproxy?stats
    maxconn 10000
    default_backend nodes443

backend nodes443
    balance roundrobin
    server web01 10.44.1.27:443 check ssl verify none
    server web02 10.44.1.26:443 check ssl verify none
    server web03 10.44.1.25:443 check ssl verify none

backend nodes80
    balance roundrobin
    server web01 10.44.1.27:80 check
    server web02 10.44.1.26:80 check
    server web03 10.44.1.25:80 check
```

Be advised of the #5311 bug.
